test/NodeOperator.test.ts

```diff
@@ -28,52 +45,77 @@ describe("NodeOperator", function () {
+        polygonERC20Contract = (await deployContract(
+            signer,
+            polygonERC20Artifact
+        )) as Polygon;
+        await polygonERC20Contract.mint(ethers.utils.parseEther("2000000"));
+
+        // deploy ERC721 token
+        const polygonERC721Artifact: Artifact =
+            await hardhat.artifacts.readArtifact("ERC721Test");
+        erc721Contract = (await deployContract(
+            signer,
+            polygonERC721Artifact
+        )) as ERC721Test;
 
         // deploy stake manager mock
-        const stakeManagerMockArtifact: Artifact = await hardhat.artifacts.readArtifact("StakeManagerMock");
-        stakeManagerMockContract = await deployContract(
+        const stakeManagerMockArtifact: Artifact =
+            await hardhat.artifacts.readArtifact("StakeManagerMock");
+        stakeManagerMockContract = (await deployContract(
             signer,
             stakeManagerMockArtifact,
-            [polygonERC20Contract.address]
-        );
-        const validatorArtifact: Artifact = await hardhat.artifacts.readArtifact("Validator");
-        validatorContract = await deployContract(
-            signer,
-            validatorArtifact,
-            []
-        );
+            [polygonERC20Contract.address, erc721Contract.address]
+        )) as StakeManagerMock;
+        const validatorArtifact: Artifact =
+            await hardhat.artifacts.readArtifact("Validator");
+        validatorContract = await deployContract(signer, validatorArtifact, []);
 
         // deploy validator factory
-        const validatorFactoryArtifact = await ethers.getContractFactory("ValidatorFactory");
-        validatorFactoryContract = await upgrades.deployProxy(
-            validatorFactoryArtifact, [validatorContract.address]);
+        const validatorFactoryArtifact: ValidatorFactory__factory =
+            (await ethers.getContractFactory(
+                "ValidatorFactory"
+            )) as ValidatorFactory__factory;
+        validatorFactoryContract = (await upgrades.deployProxy(
+            validatorFactoryArtifact,
+            [validatorContract.address]
+        )) as ValidatorFactory;
 
         // deploy node operator contract
-        const nodeOperatorRegistryArtifact = await ethers.getContractFactory("NodeOperatorRegistry");
-        nodeOperatorRegistryContract = await upgrades.deployProxy(
+        const nodeOperatorRegistryArtifact = (await ethers.getContractFactory(
+            "NodeOperatorRegistry"
+        )) as NodeOperatorRegistry__factory;
+        nodeOperatorRegistryContract = (await upgrades.deployProxy(
             nodeOperatorRegistryArtifact,
             [
                 validatorFactoryContract.address,
                 stakeManagerMockContract.address,
                 polygonERC20Contract.address
             ]
+        )) as NodeOperatorRegistry;
 
+        // deploy poLido mock contract
+        // TODO: change later with the real poLido contract.
```

Comment:
When do we need to address this TODO?
Also, poLido should be StMATIC here

Reply:
This one can be removed.
In NodeOperator we are mocking the StMATIC contract, but in StMATIC we are testing the we are not, and we tested for every case.

Reply:
@GitMark0 Please remove this TODO.